### PR TITLE
DSND-2508: Fix missing capital date bug

### PR DESCRIPTION
**Pull request by sthompsonCH**
## Describe the changes
This PR implements the fix for the capital date bug where the `date` field was not being added to the `ObjectNode` and therefore not sent in the PUT request body.

### Related Jira tickets
[DSND-2508](https://companieshouse.atlassian.net/browse/DSND-2508)

### General
- [ ] Is the PR as small as it can be?
- [ ] Has the Jira ticket been updated with any relevant comments?
- [ ] Is the `README` up to date?
- [ ] Has the `POM` been updated for the latest versions of dependencies?
- [ ] Has the code been double-checked against `main`?
- [ ] Does the code adhere standards in this repository, including SonarQube checks?

### Testing
- [ ] Do the code changes have unit and integration tests with code coverage > 80%?
- [ ] Has the code been tested locally and/or passed the API Karate tests on Tilt?
- [ ] Have mandatory manual test cases been run?
- [ ] Are extra Karate tests required?
- [ ] Are all the test data resources up to date?

### Release preparation
_Where possible, add links to the respective Jira tickets when an item is checked_
- [ ] Have these changes been included in a release ticket?
- [ ] Are changes required to the `docker-chs-development` deployment or test data?
- [ ] Are any changes required to the environment added to `chs-configs`?

## Notes to the tester
_Add any testing hints or instructions here._

[DSND-2508]: https://companieshouse.atlassian.net/browse/DSND-2508?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/filinghistory/consumer/transformrules/functions/CapitalCaptor.java [hunk 1]
```diff
@@ -45,7 +45,7 @@ CapitalCaptures captureCapital(Pattern extract, String altDescription, String so
         while (matcher.find()) {
             Map<String, Integer> namedGroups = matcher.namedGroups();
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/filinghistory/consumer/transformrules/functions/CapitalCaptor.java:46]: This does not need to be passed as a param to the private method and instead can be take straight from the matcher where needed.

**Review by cgregoryCH — APPROVED** [review]
Looks good, added a minor refactoring comment for if you've got time

**Review by crowlandsCH — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by johncook-ch — APPROVED** [review]

**Review by crowlandsCH — APPROVED** [review]

**Pull request merged**